mmv1/validate_third_party_test.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+func TestTemplatesStillNeedToBeTemplates(t *testing.T) {
+	// Get the directory where this test file is located
+	_, testFilePath, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Fatal("Failed to get current test file path")
+	}
+	testDir := filepath.Dir(testFilePath)
+
+	// Define the third_party directory relative to the test file
+	thirdPartyDir := filepath.Join(testDir, "third_party", "terraform")
+
+	// Regular expression to match Go template syntax
+	templateSyntaxRegex := regexp.MustCompile(`\{\{.*?\}\}`)
+
+	// Track files that no longer need to be templates
+	unnecessaryTemplates := []string{}
+
+	// Walk through the third_party directory
+	err := filepath.Walk(thirdPartyDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			// Handle case where third_party directory doesn't exist
+			if os.IsNotExist(err) && path == thirdPartyDir {
+				t.Logf("Warning: third_party directory not found at %s", thirdPartyDir)
+				return nil
+			}
+			return err
+		}
+
+		// Skip directories
+		if info.IsDir() {
+			return nil
+		}
+
+		// Only check .go.tmpl files
+		if filepath.Ext(path) != ".tmpl" || !strings.HasSuffix(path, ".go.tmpl") {
```

Comment:
is there a case that `filepath.Ext(path) != ".tmpl"` covers that isn't covered in `!strings.HasSuffix(path, ".go.tmpl")`?

Reply:
just one

        The following 1 .go.tmpl files in third_party directory don't contain any template syntax and no longer need to be templates:
    validate_third_party_test.go:79:   - third_party/terraform/terraform-registry-manifest.json.tmpl

Reply:
I've changed it to include all tmpl files instead of just .go.tmpl. Lmk what you think